src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -680,21 +694,57 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
 
             foreach (var project in nonBuildIntegratedProjects)
             {
-                // packages.config based projects are handled here
-                nugetActions.AddRange(await PreviewUpdatePackagesForClassicAsync(
+                var packagesToUpdateInProject = await GetPackagesToUpdateInProject(project, packageIdentities, token);
+                if (packagesToUpdateInProject.Any())
+                {
+                    var includePrerelease = packagesToUpdateInProject.Where(
+                        package => package.Version.IsPrerelease).Any();
+
+                    var updatedResolutionContext = new ResolutionContext(
+                        dependencyBehavior: resolutionContext.DependencyBehavior,
+                        includePrelease: includePrerelease,
+                        includeUnlisted: resolutionContext.IncludeUnlisted,
+                        versionConstraints: resolutionContext.VersionConstraints,
+                        gatherCache: resolutionContext.GatherCache);
+
+                    // packages.config based projects are handled here
+                    nugetActions.AddRange(await PreviewUpdatePackagesForClassicAsync(
                     packageId,
-                    packageIdentities,
+                    packagesToUpdateInProject,
                     project,
-                    resolutionContext,
+                    updatedResolutionContext,
                     nuGetProjectContext,
                     primarySources,
                     secondarySources,
                     token));
+                }
             }
 
             return nugetActions;
         }
 
+        private async Task<List<PackageIdentity>> GetPackagesToUpdateInProject(
+            NuGetProject project,
+            List<PackageIdentity> packages,
+            CancellationToken token)
+        {
+            var installedPackages = await project.GetInstalledPackagesAsync(token);
+            var packageIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
`var packageIds = new HashSet<string>(installedPackages.Select(p => p.PackageIdentity.Id), StringComparer.OrdinalIgnoreCase)`

The foreach loop below isn't needed

Reply:
I just dont trust these `LINQ` but I guess it's fine here.